acronym/acronym.go

```diff
@@ -1,15 +1,29 @@
-// This is a "stub" file.  It's a little start on your solution.
-// It's not a complete solution though; you have to write some code.
-
-// Package acronym should have a package comment that summarizes what it's about.
-// https://golang.org/doc/effective_go.html#commentary
 package acronym
 
-// Abbreviate should have a comment documenting it.
+import (
+	"strings"
+)
+
+// Abbreviate function to get the acronym of words
 func Abbreviate(s string) string {
-	// Write some code here to pass the test suite.
-	// Then remove all the stock comments.
-	// They're here to help you get started but they only clutter a finished solution.
-	// If you leave them in, reviewers may protest!
-	return ""
+
+	// replace all hyphens with space
+	spaces := strings.ReplaceAll(s, "-", " ")
+	// replace all underscors with nothing, (thus removing underscores)
+	unders := strings.ReplaceAll(spaces, "_", "")
+	// split a string into substrings, removing any space characters, including newlines
+	words := strings.Fields(unders)
+
+	res := ""
+
+	for _, word := range words {
+		// slice each word in words to give us just the individual letters
+		res += word[0:1]
+	}
+
+	// Ensure acronym is in capitals
+	acr := ""
+	acr = acr + strings.ToUpper(res)
+
+	return acr
```

Comment:
You can tidy this up a little, there isn't any need for the `acr` variable here. You already have the acronym contents in `res`. You just need to convert it to upper case:

```suggestion
	return strings.ToUpper(res)
```

Reply:
Yep, makes perfect sense. thanks :)